lib/locale_flash/flash.rb

```diff
@@ -10,11 +10,31 @@ def initialize(args={})
       @message    = args[:message]
     end
 
-    def self.from_params(type, params={})
-      if params.is_a?(Hash) && params[:controller] && params[:action]
-        new(params.merge(:type => type))
-      else
-        new(:type => type, :message => params.to_s)
+    class << self
+      def from_params(type, params={})
```

Comment:
Yeah I know we favor this form, but I never remember how we declare
private or protected method with the `def self.method` form :-(

We had this issue on at least one or two other projects, I think I
never found a solution, but will check ASAP.